### PR TITLE
prices should be rounded on propales

### PR DESCRIPTION
**Pull request by bafbes**

### CODE REVIEW

File: htdocs/comm/propal.php [hunk 1]
```diff
@@ -1996,8 +1996,8 @@
 	include DOL_DOCUMENT_ROOT . '/core/tpl/extrafields_view.tpl.php';
 
 	// Amount HT
-	print '<tr><td height="10">' . $langs->trans('AmountHT') . '</td>';
-	print '<td class="nowrap" colspan="2">' . price($object->total_ht, '', $langs, 0, - 1, - 1, $conf->currency) . '</td>';
+	print '<tr><td height="10" width="25%">' . $langs->trans('AmountHT') . '</td>';
+	print '<td class="nowrap" colspan="2">' . price($object->total_ht, '', $langs, 1, - 1, - 1, $conf->currency) . '</td>';
```
Comment (eldy) [inline htdocs/comm/propal.php:2000]: There is no reason to round here. If the total_ht is not rounded with 5 digit, it means a bug was introduced simewhere else. Data must be rounded when inserted not when output.

**Comment by bafbes** [comment]
I use a hook to intercept the "addline" action and calculate the prices on lines. The formulae used are complicated and introduce prices with many digits after the comma. While they are rounded on the line, they are not rounded in the totals. 
I saw more than five digits after the comma without having any error in the calculus.

**Comment by BebZ** [comment]
a price rounded on the line, really ? 
did you see what happens, when you change a discount on 3.8.2 or 3 ? 
you have a fixed price of 10 € let's say you reduce the discount from 5% to 2%
see what happens 
image

it is really funny because price stays the same at the end. Maybe related ? or not ?

**Comment by bafbes** [comment]
You're right. It is preferable to round the prices on the hook implementation.
Furthermore, as I didn't apply any reductions to lines, I could not see the issue reported by BebZ.
Don't this issue enforce the the proposition to round the prices on output ??

**Pull request closed without merging**